applications/dashboard/controllers/class.entrycontroller.php

```diff
@@ -468,9 +468,15 @@ public function Connect($Method) {
             $RoleIDs = $this->UserModel->NewUserRoleIDs();
          }
 
+         if (C('Garden.SSO.SyncRolesBehavior') === 'register') {
+            $SaveRolesRegister = TRUE;
+            $SaveRoles = FALSE;
+         }
+
```

Comment:
In the current behavior, the call to `Register()` below would then have 'SaveRoles' set to `true` whereas now it would be `false`.